tools/commands-packages-query.js

```diff
@@ -0,0 +1,1159 @@
+
+    // Display a warning about other release versions at the bottom.
+    if (data.totalVersions > versionsShown) {
+      var versionsPluralizer =
+            (data.totalVersions > 1) ?
+            "all " + data.totalVersions + " versions" :
+            "the hidden version";
+      Console.info(
+        "Some versions of", self.name, "have been hidden.",
+        "To see " + versionsPluralizer + ", run",
+        Console.command("'meteor show --show-all " + self.name + "'") + ".");
+    }
+  },
+  // Displays all the versions of a given release in a human-readable
+  // format. Includes experimental and otherwise hidden versions. Takes in an
+  // array of version objects, each of which has the following keys:
+  //  - version: version string
+  //  - orderKey: (optional) orderKey of this version. Not all versions have
+  //    orderKeys.
+  //  - publishedOn: date of publication
+  //  - recommended: true if the version is recommended.
+  _displayAllReleaseVersions: function (versions) {
+    var self = this;
+    var columnOpts = { indent: 2, ignoreWidth: true };
+    // If we don't have any versions, then there is nothing to display.
+    if (! versions) { return; }
+
+    // We are going to print versions with order key ('versions'), separately
+    // from versions without an order key ('experimental versions').
+    var versionsDivided = _.groupBy(versions, function (v) {
+      return _.has(v, "orderKey");
+    });
+    var experimentalVersions = versionsDivided[false];
+    var versionsWithKey = versionsDivided[true];
+
+    if (versionsWithKey) {
+      // Sort versions that have order keys by order key, so that 1.0 comes
+      // after 0.9.4.1, etc.
+      versionsWithKey = _.sortBy(versionsWithKey, function (v) {
+        return v.orderKey;
+      });
+      Console.info("Versions:");
+      var rows = [];
+      _.each(versionsWithKey, function (vr) {
+        var dateStr = utils.longformDate(vr.publishedOn);
+        if (! vr.recommended) {
+          rows.push([ vr.version, dateStr ]);
+        } else {
+          var paddedDate = padLongformDate(dateStr);
+          rows.push([ vr.version, paddedDate + "  (recommended)" ]);
+        }
+      });
+      Console.printTwoColumns(rows, columnOpts);
+    }
+
+    if (experimentalVersions) {
+      // We can't sort by order key, so sort by order of publication.
+      experimentalVersions = _.sortBy(experimentalVersions, function (v) {
+        return v.publishedOn;
+      });
+      Console.info("Experimental versions:");
+      var rows = [];
+      _.each(experimentalVersions, function (vr) {
+        // Experimental versions cannot be recommended.
+        rows.push([vr.version, utils.longformDate(vr.publishedOn)]);
+      });
+      Console.printTwoColumns(rows, columnOpts);
+    }
+  }
+});
+
+main.registerCommand({
+  name: 'show',
+  pretty: true,
+  minArgs: 1,
+  maxArgs: 1,
+  usesPackage: true,
+  options: {
+    "show-all": { type: Boolean, required: false },
```

Comment:
Wouldn't they notice the error when the option doesn't work and spits out the help anyway? We are doing a fairly large refactor, so deprecating flat-out makes sense, I think. The longer we linger with backwards-compatibility support for something like this, the harder it will be to get rid of it (and the more obnoxious the code will look)